fuel-base/src/main/kotlin/fuel/RealHttpLoader.kt

```diff
@@ -6,48 +6,108 @@ import okhttp3.Call
+    }
 
     override suspend fun post(request: Request): Response {
-        val requestBody = request.requestBody
-            ?: throw IllegalArgumentException("RequestBody should not be null")
-        val requestBuilder = Builder().headers(request.headers).post(requestBody)
-        return fetcher.fetch(request.data, requestBuilder)
+        requireNotNull(request.requestBody, { "RequestBody for method POST should not be null" })
+        return fetcher.fetch(request.data, createRequestBuilder(request, Method.POST)).await().validate()
     }
 
     override suspend fun put(request: Request): Response {
-        val requestBody = request.requestBody
-            ?: throw IllegalArgumentException("RequestBody should not be null")
-        val requestBuilder = Builder().headers(request.headers).put(requestBody)
-        return fetcher.fetch(request.data, requestBuilder)
+        requireNotNull(request.requestBody, { "RequestBody for method PUT should not be null" })
+        return fetcher.fetch(request.data, createRequestBuilder(request, Method.PUT)).await().validate()
     }
 
     override suspend fun patch(request: Request): Response {
-        val requestBody = request.requestBody
-            ?: throw IllegalArgumentException("RequestBody should not be null")
-        val requestBuilder = Builder().headers(request.headers).patch(requestBody)
-        return fetcher.fetch(request.data, requestBuilder)
+        requireNotNull(request.requestBody, { "RequestBody for method PATCH should not be null" })
+        return fetcher.fetch(request.data, createRequestBuilder(request, Method.PATCH)).await().validate()
     }
 
     override suspend fun delete(request: Request): Response {
-        val requestBuilder = Builder().headers(request.headers).delete(request.requestBody)
-        return fetcher.fetch(request.data, requestBuilder)
+        return fetcher.fetch(request.data, createRequestBuilder(request, Method.DELETE)).await().validate()
     }
 
     override suspend fun head(request: Request): Response {
-        val requestBuilder = Builder().headers(request.headers)
-        return fetcher.fetch(request.data, requestBuilder)
+        return fetcher.fetch(request.data, createRequestBuilder(request, Method.HEAD)).await().validate()
     }
 
     override suspend fun method(request: Request): Response {
-        val method = request.method ?: throw IllegalArgumentException("method should not be null")
+        val method = requireNotNull(request.method, { "method should be not null" })
         val requestBuilder = Builder().headers(request.headers).method(method, request.requestBody)
-        return fetcher.fetch(request.data, requestBuilder)
+        return fetcher.fetch(request.data, requestBuilder).await().validate()
+    }
+    //endregion
+}
+
+private fun createRequestBuilder(request: Request, method: Method): Builder {
+    return Builder().headers(request.headers).apply {
+        if (method == Method.GET || method == Method.HEAD) {
```

Comment:
We should use https://github.com/square/okhttp/blob/master/okhttp/src/main/kotlin/okhttp3/internal/http/HttpMethod.kt#L26 for better checks.

Reply:
Done